examples/contrived.rs

```diff
@@ -0,0 +1,24 @@
+use instant_segment::{Search, Segmenter};
```

Comment:
Rust has a default location for examples in the crate, so maybe we should put this in `instant-segment/examples`? Does a convention for that exist in the Python ecosystem?

Reply:
Sounds good! I don't think there are any strong conventions around that so I've moved the mirrored the directory structure in the Python directory.